src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,336 @@
+        result += "Now you have " + tasks.size() + (tasks.size() > 1 ? " tasks" : " task") + " in the list.";
+        result += "\n";
+        result += Ui.showLine();
+        return result;
+    }
+
+    /**
+     * Adds a task of type Event to the task list and updates the storage (file).
+     *
+     * @param message The description and the event duration of the task.
+     * @param tasks The task list for the addition of the task.
+     * @param storage The storage object for that contains the file for updating.
+     * @return The response of the bot after adding an event.
+     * @throws UnmatchedArgumentException If the arguments passed in are insufficient.
+     * @throws EmptyDescriptionException If the description of the task is empty.
+     * @throws DateTimeException If the start or end time of the event is invalid.
+     */
+    public String addEvent(String message, TaskList tasks, Storage storage) throws UnmatchedArgumentException, EmptyDescriptionException {
+
+        String[] arr = message.split(" /");
+        Parser p = new Parser();
+        if (arr.length < 3) {
+            throw new UnmatchedArgumentException(arr.length, 3);
+        }
+        if (arr[0].isBlank()) {
+            throw new EmptyDescriptionException("todo");
+        }
+        LocalDateTime start = p.checkDateAndTime(arr[1].substring(5).replace(" ", "/"));
+        LocalDateTime end = p.checkDateAndTime(arr[2].substring(3).replace(" ", "/"));
+        if (end.isBefore(start) || end.isEqual(start)) {
+            throw new DateTimeException("The end is date should not be earlier or the same as the start date.");
+        }
+        Event e = new Event(arr[0], false, start, end);
+        tasks.add(e);
+        storage.writeInto(tasks);
+        String result =  Ui.showLine() + "\n" + "Got it. I've added this task: " + "\n" + " " + e + "\n";
+        result += "Now you have " + tasks.size() + (tasks.size() > 1 ? " tasks" : " task") + " in the list.";
+        result += "\n";
+        result += Ui.showLine();
+        return result;
+    }
+
+    /**
+     * Deletes a task from the task list based on the index provided and updates the storage (file).
+     *
+     * @param index The task with the index to be deleted.
+     * @param tasks The task list for the deletion of task.
+     * @param storage The storage object that contains the file for updating.
+     * @return The response of the bot when the task is deleted.
+     */
+    public String delete(int index, TaskList tasks, Storage storage) {
+
+        tasks.remove(index - 1);
+        storage.writeInto(tasks);
+        String result = Ui.showLine() + "\n" + "Noted. I've removed this task:" + "\n  " + tasks.get(index - 1);
+        result += "\nNow that you have " + tasks.size() + (tasks.size() < 2 ? " task" : " tasks") + " in the list.";
+        result += "\n";
+        result += Ui.showLine();
+        return result;
+    }
```

Comment:
Do consider putting the commands into sub classes! eg.DeleteCommand (makes it easier for you to find and edit next time too)